doc/gazebo_simulation/gazebo_simulation.rst

```diff
@@ -0,0 +1,351 @@
+    <xacro:panda_arm xyz="0 0 0" rpy="0 0 0" connected_to="world"/>
+
+It alone doesn't fix the problem, since now we need to provide a link with name :code:`world`. Open the 
+:code:`franka_description/robots/panda_arm.xacro` file and add:
+
+.. code-block:: xml
+
+    <link name="world" />
+
+between lines 3 and 4. Additionally, we should rename the fixed joint to :code:`virtual_joint` to properly match
+the SRDF specification created in previous tutorial.
+
+2. Add damping to the joint specifications
+------------------------------------------
+In this step, we need to add:
+
+.. code-block:: xml
+
+    <dynamics damping="1.0"/>
+
+to every joint (except :code:`${arm_id}_joint8}` joint) in :code:`franka_description/robots/panda_arm.xacro` file.
+
+
+3. Add inertia matrices and masses to the links
+-----------------------------------------------
+Inertia matrices are required for Gazebo physics engine to work properly. We don't require exact numbers for pedagogic reasons, 
+yet it is perfectly suitable to go after them with MeshLab as explained at `relevant Gazebo documentation page <http://gazebosim.org/tutorials?tut=inertia&cat=build_robot>`_
+in great depth. Add following snippet to the links from :code:`${arm_id}_link0` to :code:`${arm_id}_link8` in :code:`franka_description/robots/panda_arm.xacro`:
+
+.. code-block:: xml
+
+    <inertial>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <mass value="3.06" />
+        <inertia ixx="0.3" ixy="0.0" ixz="0.0" iyy="0.3" iyz="0.0" izz="0.3" />
+    </inertial>
+
+
+Similarly add following snippet to the :code:`${ns}_hand` link in :code:`franka_description/robots/hand.xacro` file:
+
+.. code-block:: xml
+
+    <inertial>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <mass value="0.68"/>
+        <inertia ixx="0.1" ixy="0.0" ixz="0.0" iyy="0.1" iyz="0.0" izz="0.1"/>
+    </inertial>
+
+Finally add following inertia block to :code:`finger` links:
+
+.. code-block:: xml
+
+    <inertial>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <mass value="0.01"/>
+        <inertia ixx="0.1" ixy="0.0" ixz="0.0" iyy="0.1" iyz="0.0" izz="0.1"/>
+    </inertial>
+
+As previously mentioned, these values come from the referred blog post. It is explicitly advised to have a look in there to grasp the matter in-depth.
+
+4. Add friction and colorize the links
+--------------------------------------
+In order to have a nice illustration of the robot in Gazebo simulation we need to colorize the links. 
+Moreover friction forces are added in order to have realistic dynamics. You can ignore them at all or change their values to experiment.
+Since the focus is MoveIt in this tutorial, we will just use the values from the provided solution.
+
+This step is a bit tedious to do manually, so the ultimate :code:`xacro` file is provided entirely in below:
+
+.. code-block:: xml
+
+    <!-- panda.gazebo.xacro -->
+    <?xml version='1.0' encoding='utf-8'?>
+    <robot xmlns:xacro="http://www.ros.org/wiki/xacro">
```

Comment:
By a blind copy of the code-block  (as I did) you get an error during the urdf generation from xacro file.
Xacro file have to start with `<?xml version='1.0' encoding='utf-8'?>`. 
I suggest therefore to swap line 1 with line 2. In this way we still have the information on the file name which is very useful, but the copy and paste will work.